### ISSUE TITLE
1.11.0-alpha2: problems with precompilation

### ISSUE THREAD

**Issue by matthias314** [JuliaLang/julia#53831]
This may be related to #52283. 

My `startup.jl` contains the line `using InteractiveUtils`. Starting Julia 1.11.0-alpha2 with `--startup-file=no` works without problems. However, without that option I get
```
~$ /usr/local/julia-1.11.0-alpha2/bin/julia
┌ Info: Precompiling InteractiveUtils [b77e0a4c-d291-57a0-90e8-8db25a27a240] (cache misses: invalid header (10))
└ @ Base loading.jl:2845
┌ Warning: The call to compilecache failed to create a usable precompiled cache file for Markdown [d6f4376e-aef5-505a-96c1-9c027394607a]
│   exception = ErrorException("Required dependency Base.PkgId(Base.UUID(\"2a0f44e3-6c83-55bd-87e4-b1978d98bd5f\"), \"Base64\") failed to load from a cache file.")
└ @ Base loading.jl:2365
┌ Info: Skipping precompilation due to precompilable error. Importing InteractiveUtils [b77e0a4c-d291-57a0-90e8-8db25a27a240].
│   exception = nothing
└ @ Base loading.jl:2356
┌ Info: Precompiling Markdown [d6f4376e-aef5-505a-96c1-9c027394607a] (cache misses: wrong source (1), invalid header (8))
└ @ Base loading.jl:2845
┌ Warning: The call to compilecache failed to create a usable precompiled cache file for Markdown [d6f4376e-aef5-505a-96c1-9c027394607a]
│   exception = ErrorException("Required dependency Base.PkgId(Base.UUID(\"2a0f44e3-6c83-55bd-87e4-b1978d98bd5f\"), \"Base64\") failed to load from a cache file.")
└ @ Base loading.jl:2365
┌ Info: Precompiling Base64 [2a0f44e3-6c83-55bd-87e4-b1978d98bd5f] (cache misses: wrong source (1), invalid header (8))
└ @ Base loading.jl:2845
┌ Warning: Replacing module `Base64`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `Markdown`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `InteractiveUtils`
└ @ Base loading.jl:2222
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-alpha2 (2024-03-18)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> 
```
It repeats every time I start Julia. The v1.11 project is empty:
```
(@v1.11) pkg> st
Status `/usr/local/julia/environments/v1.11/Project.toml` (empty project)
```
I didn't try previous versions of v1.11.0. With master I similarly get
```
$ ./julia
┌ Info: Precompiling InteractiveUtils [b77e0a4c-d291-57a0-90e8-8db25a27a240] (cache misses: invalid header (2))
└ @ Base loading.jl:2845
┌ Warning: Replacing module `Base64`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `Markdown`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `InteractiveUtils`
└ @ Base loading.jl:2222
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.12.0-DEV.231 (2024-03-22)
 _/ |\__'_|_|_|\__'_|  |  Commit d68a04ee9c (1 day old master)
|__/                   |

julia> 

$ ./julia
┌ Warning: Replacing module `Base64`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `Markdown`
└ @ Base loading.jl:2222
┌ Warning: Replacing module `InteractiveUtils`
└ @ Base loading.jl:2222
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.12.0-DEV.231 (2024-03-22)
 _/ |\__'_|_|_|\__'_|  |  Commit d68a04ee9c (1 day old master)
|__/                   |

julia> 
```
```
Julia Version 1.11.0-alpha2
Commit 9dfd28ab751 (2024-03-18 20:35 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 4 × Intel(R) Core(TM) i3-10110U CPU @ 2.10GHz
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, skylake)
Threads: 1 default, 0 interactive, 1 GC (on 4 virtual cores)
Environment:
  JULIA_DEPOT_PATH = /usr/local/julia
```

**Comment by matthias314** [comment]
UPDATE: It seems the same problem occurs without startup file:
```
$ /usr/local/julia-1.11.0-alpha2/bin/julia --startup-file=no
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-alpha2 (2024-03-18)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.11) pkg> add Chairmarks
   Resolving package versions...
    Updating `/usr/local/julia/environments/v1.11/Project.toml`
  [0ca39b1e] + Chairmarks v1.2.0
    Updating `/usr/local/julia/environments/v1.11/Manifest.toml`
  [0ca39b1e] + Chairmarks v1.2.0
  [de0858da] + Printf v1.11.0
  [4ec0a83e] + Unicode v1.11.0
Precompiling project...
  ✓ Unicode
  ✓ Printf
  3 dependencies successfully precompiled in 4 seconds
  2 dependencies precompiled but different versions are currently loaded. Restart julia to access the new versions

(@v1.11) pkg> [Ctrl-D]
$ /usr/local/julia-1.11.0-alpha2/bin/julia --startup-file=no
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-alpha2 (2024-03-18)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Chairmarks
Precompiling Chairmarks...
  ✓ Unicode
  3 dependencies successfully precompiled in 3 seconds
  1 dependency precompiled but a different version is currently loaded. Restart julia to access the new version
┌ Info: Precompiling Chairmarks [0ca39b1e-fe0b-4e98-acfc-b1656634c4de] 
└ @ Base loading.jl:2845
┌ Warning: The call to compilecache failed to create a usable precompiled cache file for Chairmarks [0ca39b1e-fe0b-4e98-acfc-b1656634c4de]
│   exception = ErrorException("Required dependency Base.PkgId(Base.UUID(\"de0858da-6303-5e67-8744-51eddeeeb8d7\"), \"Printf\") failed to load from a cache file.")
└ @ Base loading.jl:2365
Precompiling Printf...
  ✓ Unicode
  2 dependencies successfully precompiled in 1 seconds
  1 dependency precompiled but a different version is currently loaded. Restart julia to access the new version
┌ Info: Precompiling Printf [de0858da-6303-5e67-8744-51eddeeeb8d7] (cache misses: wrong source (2), invalid header (12), mismatched flags (2))
└ @ Base loading.jl:2845
```

**Comment by matthias314** [comment]
Any thoughts on this? The failing precompilation makes Julia 1.11 practically unusable for me.

**Comment by matthias314** [comment]
closed in favor of #53983
